packages/jellyfish-api-core/__tests__/category/icxorderbook/common.test.ts

```diff
@@ -0,0 +1,204 @@
+  tokenInfo = await container.call('gettoken', [symbolDFI])
+  idDFI = Object.keys(tokenInfo)[0]
+
+  // send funds to accounts
+  const payload: { [key: string]: string } = {}
+  payload[accountDFI] = '500@' + symbolDFI
+  payload[accountBTC] = '10@' + symbolDFI // for fee
+  await container.call('utxostoaccount', [payload])
+
+  // create pool
+  poolOwner = await container.call('getnewaddress', ['', 'legacy'])
+  await accountToAccount(container, symbolBTC, 1, { from: accountBTC, to: accountDFI })
+
+  const poolPairMetadata = {
+    tokenA: idBTC,
+    tokenB: idDFI,
+    commission: 1,
+    status: true,
+    ownerAddress: poolOwner,
+    pairSymbol: 'BTC-DFI'
+  }
+  await container.call('createpoolpair', [poolPairMetadata, []])
+  await container.generate(1)
+
+  // check
+  const pool = await container.call('getpoolpair', ['BTC-DFI'])
+  const combToken = await container.call('gettoken', ['BTC-DFI'])
+  const idDFIBTC = Object.keys(combToken)[0]
+  expect(pool[idDFIBTC].idTokenA).toBe(idBTC)
+  expect(pool[idDFIBTC].idTokenB).toBe(idDFI)
+
+  // add pool liquidity
+  const poolLiquidityMetadata: { [key: string]: string [] } = {}
+  poolLiquidityMetadata[accountDFI] = ['1@' + symbolBTC, '100@' + symbolDFI]
+
+  await container.call('addpoolliquidity', [poolLiquidityMetadata, accountDFI, []])
+  await container.generate(1)
+  DEX_DFI_PER_BTC_RATE = 100
+
+  const accountInfo = await container.call('getaccount', [accountDFI, {}, true])
+  expect(Object.keys(accountInfo).length).toBe(2)
+
+  // set taker fee rate (in BTC)
+  await container.call('setgov', [{ ICX_TAKERFEE_PER_BTC: 0.001 }])
+  await container.generate(1)
+  const result: any = await container.call('getgov', ['ICX_TAKERFEE_PER_BTC'])
+  expect(result.ICX_TAKERFEE_PER_BTC as number).toStrictEqual(0.001)
+  ICX_TAKERFEE_PER_BTC = result.ICX_TAKERFEE_PER_BTC as number
+}
+
+export async function checkDFISellOrderDetails (container: MasterNodeRegTestContainer, orderInput: ICXOrder, createOrderTxId: string,
+  orderInfoRetrived: Record<string, ICXOrderInfo>): Promise<void> {
+  expect(orderInfoRetrived[createOrderTxId].chainTo).toStrictEqual(orderInput.chainTo)
+  expect(orderInfoRetrived[createOrderTxId].tokenFrom).toStrictEqual(orderInput.tokenFrom === idDFI ? symbolDFI : symbolBTC)
+  expect(orderInfoRetrived[createOrderTxId].ownerAddress).toStrictEqual(orderInput.ownerAddress)
+  expect(orderInfoRetrived[createOrderTxId].receivePubkey).toStrictEqual(orderInput.receivePubkey)
+  expect(orderInfoRetrived[createOrderTxId].amountFrom).toStrictEqual(orderInput.amountFrom)
+  expect(orderInfoRetrived[createOrderTxId].amountToFill).toStrictEqual(orderInput.amountFrom)
+  expect(orderInfoRetrived[createOrderTxId].orderPrice).toStrictEqual(orderInput.orderPrice)
+  expect(orderInfoRetrived[createOrderTxId].amountToFillInToAsset).toStrictEqual(orderInput.amountFrom.multipliedBy(orderInput.orderPrice))
+  // NOTE(surangap): sometimes received == expected - 1 ??
```

Comment:
Here sometimes received == expected - 1 , i.e `orderInfoRetrived[createOrderTxId].expireHeight` is short by 1 sometimes.